pkg/proxy/iptables/proxier_test.go

```diff
@@ -419,17 +419,44 @@ func NewFakeProxier(ipt utiliptables.Interface) *Proxier {
 	return p
 }
 
-func countRules(table, data string) int {
-	inRightTable := false
-	rules := 0
-	for _, line := range strings.Split(data, "\n") {
-		if line == "" {
-			continue
-		}
-		if line[0] == '*' {
-			inRightTable = (line == "*"+table)
+// parseIPTablesData takes iptables-save output and returns a map of table name to array of lines.
+func parseIPTablesData(ruleData string) (map[string][]string, error) {
+	rawTables := strings.Split(strings.TrimPrefix(ruleData, "\n"), "COMMIT\n")
+	nTables := len(rawTables) - 1
+	if nTables < 2 || rawTables[nTables] != "" {
+		return nil, fmt.Errorf("bad ruleData (%d tables)\n%s", nTables, ruleData)
+	}
+
+	tables := make(map[string][]string, nTables)
+	for i, table := range rawTables[:nTables] {
+		lines := strings.Split(strings.Trim(table, "\n"), "\n")
```

Comment:
interesting how complex is parsing this :smile:  https://go.dev/play/p/hMn54GBM5O7